REVIEW

eadwu opened the pull request.

eadwu committed:
nixpkgs-fmt

eadwu committed:
git: ignore nix outputs

eadwu committed:
flake.nix: update description

eadwu committed:
flake.nix: update sources

eadwu committed:
thunderbird-extensions.tbsync: init

eadwu committed:
thunderbird-with-extensions: init

eadwu committed:
flake.nix: explicitly set ref for tbsync-src

eadwu committed:
github: integrate Github Actions

eadwu committed:
flake.lock: init

eadwu committed:
github: use nixFlakes from nixpkgs

eadwu committed:
flake.nix: fix tbsync reference

eadwu committed:
flake.nix: include VM tests

eadwu committed:
modules/thunderbird: init

eadwu committed:
github: run flake checks

eadwu committed:
thunderbird-extensions.dav-4-tbsync: init at 1.12

eadwu committed:
thunderbird-extensions.eas-4-tbsync: init at 1.14

eadwu committed:
Revert "github: run flake checks"

This reverts commit 1dfea2c44eed2bb7e40e69a887cb76a87a5af9fa.

The machine used doesn't seem to be supported.
  error: a 'x86_64-linux' with features {kvm, nixos-test} is required to build '/nix/store/rcnffq6qs0dg2in42x84a9gmcrfhz4yz-vm-test-run-unnamed.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, nixos-test}

eadwu committed:
thunderbird-utils: init

Convert existing extensions to use the utilities

eadwu committed:
thunderbird-extensions.enigmail: init at 2.1.6

eadwu committed:
flake.nix: sync up coverage of new derivations

eadwu committed:
flake.nix: fix evaluation of vmTests

eadwu committed:
REVIEW.md: include a basic sample review

eadwu committed:
thunderbird-with-extensions: disable app updates

eadwu committed:
thunderbird-utils.buildThunderbirdExtension: remove unnecessary directory

eadwu committed:
thunderbird-utils.buildThunderbirdExtension: allow custom buildPhase

eadwu committed:
thunderbird-extensions.autocrypt: init

eadwu committed:
flake.lock: Update

Flake input changes:

* Added 'autocrypt-src': 'github:autocrypt-thunderbird/autocrypt-thunderbird/3953dda07ad8b0d99beab1c9e61da147bcaf3ff5'

eadwu committed:
github: rename job name

eadwu committed:
github: use a less aggressive cron schedule

eadwu committed:
flake.nix: convert tests to stdenv-based

Files are copied locally so network isn't needed
Should also run fine on hydra now

eadwu committed:
github: run checks as well

Since they aren't VM-based now might as well

rolfschr commented on REVIEW.md:
```diff
@@ -0,0 +1,38 @@
+A basic overview of the what should be reviewed.
```
Perfect, thanks a lot for this! I guess you could remove this file afterwards (or maybe not?)

rolfschr commented on REVIEW.md:
```diff
@@ -0,0 +1,38 @@
+- A visual reference can be found at https://github.com/jobisoft/TbSync/wiki/How-to-get-started
+- An initial setup is not needed, feel free to just press cancel and move on
+- Click the calendar icon on the top right and verify that it should be empty (as it should by default)
+- Click the hamburger icon next on the menubar near the `Tasks` section and click through `Add-ons` > `TbSync`
```
Didn't find this. I found ``Edit`` -> ``Synchronization Settings (TbSync)``, though.

rolfschr commented on REVIEW.md:
```diff
@@ -0,0 +1,38 @@
+- With the new window click `Account actions` > `Add new account` > `CalDAV & CardDAV`
+- With the new window click `Google` > `Next` > `Next` and login to the Gmail account and allowing TbSync access to the account for the permissions requested before clicking `Finish`
+- Check `Enable and Synchronize this account` and wait for the `Status` and `Avaiable resources` sections to pop up and populate
+- Check the checkbox next to the calendar icon(s) before clicking `Synchronize now` on the bottom right
```
I didn't get the calendar but the "address book" resource and synchronizing went fine.

rolfschr commented on REVIEW.md:
```diff
@@ -0,0 +1,38 @@
+- Exit the TbSync settings tab back to the main window and switch to the `Calendar` tab where a "Calendar" should have popped up with the name `Google (your_email_here@gmail.com)` or `Google ...` if there were multiple calendars
+- If there were any events in your calendar at Gmail, it should also pop up
+
+### Test Script
```
Worked like a charm!

rolfschr commented on default.nix:
```diff
@@ -1,3 +1,4 @@
   src = builtins.fetchGit ./.;
-}).defaultNix
+}
+).defaultNix
```
I run ``nixfmt`` on your branch and got quite a bit of changes. Not sure they should be incorporated. Your call.

rolfschr commented on pkgs/thunderbird-extensions/default.nix:
```diff
@@ -0,0 +1,16 @@
+    pname = "enigmail";
+    version = "2.1.6";
+
+    id = "71";
```
Unused?

eadwu replied:
Couldn't get the Makefile in the repository to build correctly (if I remember correctly) so this is directly fetched from mozilla, which has a component in the path (not quite sure if it's the addon id but that's what I called it).

rolfschr commented on pkgs/thunderbird/with-extensions.nix:
```diff
@@ -0,0 +1,53 @@
+  dontStrip = true;
+  dontPatchELF = true;
+  meta = thunderbird.meta // {
+    platforms = lib.platforms.linux;
```
Only for Linux?

eadwu replied:
Upstream restricted Thunderbird to `platforms.linux` so that's why I copied it, but I'll switch to `platforms.all` since the wrapper in itself should work fine on all platforms (though it'll still fail on the said platforms due to the dependency).

rolfschr commented on flake.nix:
```diff
@@ -23,96 +19,133 @@
+              sleep 128
+              ls $HOME/fakeprofile/extensions | grep -q ${tbsync.emid} && exit 1 || echo "valid extensions"
+              [ "$(ls $HOME/fakeprofile/extensions | wc -l)" = "1" ] && echo "valid number of extensions" || exit 1
+            '';
```
I think I'm missing something here: Why would you expect the tbsync extension if the "default" thunderbird?

eadwu replied:
Thunderbird includes an extension by default (Lighting - Integrated Calendar).

rolfschr reviewed (commented):
LGTM. I have a couple of comments/questions but only nitpicks (or even misunderstandings on my side I believe). Looks like you had to dig into this quite a bit!

eadwu committed:
thunderbird-with-extensions: allow wrapper on all platforms

Expected to fail outside of linux due to the dependency on Thunderbird

See https://github.com/ngi-nix/thunderbird-extensions/pull/2#discussion_r472849886

eadwu committed:
nixpkgs-fmt

eadwu committed:
REVIEW.md: incorporate changes through #2

Specifically see the following:
https://github.com/ngi-nix/thunderbird-extensions/pull/2#discussion_r472811451
https://github.com/ngi-nix/thunderbird-extensions/pull/2#discussion_r472815541

rolfschr commented:
Good stuff! :+1:
